### ISSUE TITLE
Add support for different community plugins

### ISSUE THREAD

**Issue by dianjuar** [bikecoders/ngx-deploy-npm#415]
We currently offer support to all official library generators (Nx Plugins). Let's expand to cover the community Nx Plugin, such as:

- [x] #437
- [x] #438
- [x] #441
- [x] #440
- [x] #439
- [x] #442
- [x] #443
- [x] #444

By offer support, I mean to ngx-deploy-npm work right off the bat, immediately, without any configurations, specifically on the `dist` folder detection. They can use [`--dist-folder-path`](https://github.com/bikecoders/ngx-deploy-npm#--dist-folder-path) if we cannot determine it, but we would provide a better experience if they don't have to add an extra configuration to be able to publish.

Add the supported Plugins to our [integration test](https://github.com/bikecoders/ngx-deploy-npm/tree/main/e2e/ngx-deploy-npm-e2e/integration) and in our package.json dev dependencies to be able to track new versions and continuously test library publishment with their plugins.

We might have full support now. In that case, awesome! but we still want to have them on our integration test.

**Comment by dianjuar** [comment]
For Contributors:

# TL; DR

1. Verify if the plugin is currently supported
    - Unlike projects, libraries could have different methods to indicate the path of the dist folder.
    - We might have support already for the community nx plugin. With support, we meant that `ngx-deploy-npm` already has a strategy to locate the dist folder path.
    - You can check if this project already offers support by:
      - **(Recommended)** creating a library and trying to publish it to NPM
      - **(Hackerman)** verifying the current project [strategies](https://github.com/bikecoders/ngx-deploy-npm/tree/main/packages/ngx-deploy-npm/src/executors/deploy/utils/get-lib-output-path/strategies)
2. Create a strategy to locate the community dist folder path (only if needed)
3. Add the technology badge to our [Readme](https://github.com/bikecoders/ngx-deploy-npm#supported-library-generators) 
4. Create an [integration test](https://github.com/bikecoders/ngx-deploy-npm/tree/main/e2e/ngx-deploy-npm-e2e/integration)
5. Create a PR

---

# 1. Verify if the plugin is currently supported 🕵️🕵️‍♀️ 

Unlike projects, libraries could have different methods to indicate the path of the dist folder. We might already have support for the community nx plugin you are interested in. With support, we meant that `ngx-deploy-npm` already has a strategy to locate the dist folder path.

The current strategies are: 
- [`outputPath` build option](https://github.com/bikecoders/ngx-deploy-npm/blob/main/packages/ngx-deploy-npm/src/executors/deploy/utils/get-lib-output-path/strategies/output-path-build-option/output-path-build-option.ts#L7-L26)
- [`ng-package.json` file](https://github.com/bikecoders/ngx-deploy-npm/blob/main/packages/ngx-deploy-npm/src/executors/deploy/utils/get-lib-output-path/strategies/ng-package/ng-package.ts#L9-L25)
- [`--dist-folder-path` option](https://github.com/bikecoders/ngx-deploy-npm/blob/main/packages/ngx-deploy-npm/src/executors/deploy/utils/get-lib-output-path/strategies/dist-folder-path/dist-folder-path-option.ts#L7-L16)

If the community plugin applies one of those strategies, you are ready with step one ✅ 

You can also verify if the community plugin is supported by creating an Nx Workspace and trying to publish a library created with that community plugin. For that, you need the following:

1. Create an Nx Workspace
    - You can use our tool by typing on the console [`./tools/create-nx-workspace.sh`](https://github.com/bikecoders/ngx-deploy-npm/blob/main/tools/create-nx-workspace.sh)
2. Create a library with the Nx Plugin
3. Configure `ngx-deploy-npm` on the new lib
    - `npx nx generate ngx-deploy-npm:install --projects YOUR_LIB`
4. Try to publish it
    - `npx nx deploy YOUR_LIB --dry-run`

If you received the error `ngx-deploy-npm was not able to detect the dist path.` we, unfortunately, don't support that community plugin 😭. You would need to apply step two of this journey.

# 2. Create the strategy 👨‍💻

If `ngx-deploy-npm` doesn't have a proper strategy to locate the dist folder path of your community plugin, you would need to create one.

There are two main steps here:

1. Create the [strategy](https://github.com/bikecoders/ngx-deploy-npm/tree/main/packages/ngx-deploy-npm/src/executors/deploy/utils/get-lib-output-path/strategies) itself
2. Make our [strategy chooser](https://github.com/bikecoders/ngx-deploy-npm/blob/main/packages/ngx-deploy-npm/src/executors/deploy/utils/get-lib-output-path/strategy-selector.ts#L16) smarter by codifying when is the right moment to apply your strategy.

# 3. Add the technology supported to our Readme 📚 

Let the world know that ngx-deploy-npm now supports your technology by listing it on the [README](https://github.com/bikecoders/ngx-deploy-npm#supported-library-generators)

4. Create an Integration Test ⚙️ 

When we say to "offer support", we mean it. We want to ensure that ngx-deploy-npm still works for future versions of that community plugin. To accomplish that, we must create an Integration Test.

This one is easy. You need to:

1. Install the Nx Community Plugin in our devDependcies
   - Dependabot would help us to run that integration test whenever a new version comes out
2. Create an [Integration Test](https://github.com/bikecoders/ngx-deploy-npm/tree/main/e2e/ngx-deploy-npm-e2e/integration ) among the other supported technologies

5. Create the PR 🎉 

This would be the end of the journey. You are finally ready to create a PR to offer support to your loved community plugin and make it possible to publish that library with only one command. We are so lucky to have you 💪 

---

Useful resources:
- [Contributors Readme](https://github.com/bikecoders/ngx-deploy-npm/blob/main/docs/README_contributors.md)

**Comment by dianjuar** [comment]
This feature has been deprecated on version 8.0.0.
